providers/providers.go

```diff
@@ -310,6 +314,21 @@ func (pm *ProviderManager) AddProvider(ctx context.Context, k []byte, val peer.I
 	}
 }
 
+func (pm *ProviderManager) addProv(k []byte, p peer.ID) error {
+	now := time.Now()
+	// If there is something on cache, update cache
+	if cached, ok := pm.cache.Get(string(k)); ok {
+		pSet := cached.(*providerSet)
+		pSet.providers = append(pSet.providers, p)
+		pSet.set[p] = now
+
+		// Note to dev: .Add does replace, not merge
+		pm.cache.Add(string(k), &(*pSet))
```

Comment:
Ahhh pointers...is this equivalent to:

```suggestion
		pm.cache.Add(string(k), pSet)
```

???

Reply:
AFAIK, only `*(&pSet)` would be equivalent to `pSet`. `&(*pSet)` is the address of thing that pSet is pointing to